sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs

```diff
@@ -113,47 +110,31 @@ public virtual async Task<Response<Secret>> SetAsync(Secret secret, Cancellation
         {
             if (secret == null) throw new ArgumentNullException(nameof(secret));
 
-            if (secret.Name == null) throw new ArgumentNullException($"{nameof(secret)}.{nameof(secret.Name)}");
-
-            if (secret.Value == null) throw new ArgumentNullException($"{nameof(secret)}.{nameof(secret.Value)}"); ;
-
             return await SendRequestAsync(HttpPipelineMethod.Put, secret, () => new Secret(), cancellationToken, SecretsPath, secret.Name);
         }
 
         public virtual Response<Secret> Set(Secret secret, CancellationToken cancellationToken = default)
         {
             if (secret == null) throw new ArgumentNullException(nameof(secret));
 
-            if (secret.Name == null) throw new ArgumentNullException($"{nameof(secret)}.{nameof(secret.Name)}");
-
-            if (secret.Value == null) throw new ArgumentNullException($"{nameof(secret)}.{nameof(secret.Value)}");
-
             return SendRequest(HttpPipelineMethod.Put, secret, () => new Secret(), cancellationToken, SecretsPath, secret.Name);
         }
 
         public virtual async Task<Response<Secret>> SetAsync(string name, string value, CancellationToken cancellationToken = default)
         {
-            if (string.IsNullOrEmpty(name)) throw new ArgumentNullException(nameof(name));
-
-            if (string.IsNullOrEmpty(value)) throw new ArgumentNullException(nameof(value));
-
             return await SetAsync(new Secret(name, value), cancellationToken);
         }
 
         public virtual Response<Secret> Set(string name, string value, CancellationToken cancellationToken = default)
         {
-            if (string.IsNullOrEmpty(name)) throw new ArgumentNullException(nameof(name));
-
-            if (string.IsNullOrEmpty(value)) throw new ArgumentNullException(nameof(value));
-
             return Set(new Secret(name, value), cancellationToken);
         }
 
         public virtual async Task<Response<DeletedSecret>> DeleteAsync(string name, CancellationToken cancellationToken = default)
         {
             if (string.IsNullOrEmpty(name)) throw new ArgumentNullException(nameof(name));
 
-            return await SendRequestAsync(HttpPipelineMethod.Delete, () => new DeletedSecret(), cancellationToken, SecretsPath, name);
+            return await SendRequestAsync(HttpPipelineMethod.Delete, () => new DeletedSecret(), cancellationToken, SecretsPath, name).ConfigureAwait(false);
```

Comment:
why are we adding `.ConfigureAwait(false)`

Reply:
I'll revert this change. Part of trying to debug the hang of full framework.